lte/gateway/python/magma/pipelined/app/restart_mixin.py

```diff
@@ -0,0 +1,238 @@
+"""
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+from typing import List
+from abc import ABCMeta, abstractmethod
+
+from ryu.ofproto.ofproto_v1_4_parser import OFPFlowStats
+
+from lte.protos.pipelined_pb2 import SetupFlowsResult, ActivateFlowsRequest
+from magma.pipelined.app.base import ControllerNotReadyException
+from magma.pipelined.openflow import flows
+from magma.policydb.rule_store import PolicyRuleDict
+
+from magma.pipelined.policy_converters import FlowMatchError, \
+    convert_ipv4_str_to_ip_proto, ovs_flow_match_to_magma_match
+
+
+class RestartMixin(metaclass=ABCMeta):
+    """
+    RestartMixin
+
+    Mixin class for policy enforcement apps that includes common methods
+    used for rule activation/deactivation.
```

Comment:
It would be nice to extend this functionality to all controllers.

Reply:
yup will work on that